src/auth/basic/SASL/required.m4

```diff
@@ -6,4 +6,9 @@
 ##
 
 SQUID_CHECK_SASL
-AS_IF([test "x$squid_cv_check_sasl" = "xyes"],[BUILD_HELPER="SASL"])
+AS_IF([test "x$squid_cv_check_sasl" = "xyes"],[
+    BUILD_HELPER="SASL"
+    AS_IF([test "x$squid_host_os" = "xdarwin"],[
+        SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED([-Wno-deprecated-declarations])
```

Comment:
If both SASL and LDAP are enabled on MacOS, how many `-Wno-deprecated-declarations` flags will be added to CXXFLAGS? If the answer is "more than one", then we need to refactor so that the answer becomes "at most one".

The exact solution will depend on the other change request. For example, if the second bullet in that change request applies, then we probably need to place a single, conditional SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED() call _after_ all LDAP and SASL checks.

Reply:
It wasn't happening in my tests, but I've added code to explicitly avoid duplication